src/main/setupCheckFirstRun.js

```diff
@@ -0,0 +1,38 @@
+import fs from 'fs';
+import path from 'path';
+
+import {
+  app,
+  ipcMain,
+} from 'electron';
+
+import { ipcMainChannels } from './ipcMainChannels';
+import { getLogger } from '../logger';
+
+const logger = getLogger(__filename.split('/').slice(-1)[0]);
+
+export const APP_HAS_RUN_TOKEN = 'app-has-run-token';
+
+/** Check if the user has run this application before.
+ *
+ * @returns {boolean}
+ */
+export function checkFirstRun() {
+  const userDataPath = app.getPath('userData');
+  const hasRunTokenPath = path.join(userDataPath, APP_HAS_RUN_TOKEN);
+  try {
+    if (fs.existsSync(hasRunTokenPath)) {
+      return false;
+    }
+    fs.writeFileSync(hasRunTokenPath, '');
+  } catch (error) {
+    logger.warn(`Unable to write first-run token: ${error}`);
+  }
+  return true;
+}
+
+export function setupCheckFirstRun() {
+  ipcMain.handleOnce(
+    ipcMainChannels.IS_FIRST_RUN, (event) => checkFirstRun()
```

Comment:
I found [this answer](https://stackoverflow.com/a/47084018/14451410) which says you can call any javascript function with any number of arguments. So I imagine this is fine to stay as it is. If you want to change them to `() => checkFirstRun()`, that would make sense, since no arguments are being used.

Reply:
Cool, thanks for the reference! I think I'll change them to `() => { }` since having an unused argument seems undesirable too. Especially if this were a much longer function, it wouldn't immediately be obvious that `event` was not used.